crates/rolldown_plugin_json/src/lib.rs

```diff
@@ -25,27 +34,62 @@ impl Plugin for JsonPlugin {
       return Ok(None);
     }
     let code = strip_bom(args.code);
-    let value = serde_json::from_str::<Value>(code)?;
-    if self.stringify {
-      let normalized_code = if self.is_build {
-        let str = serde_json::to_string(&value)?;
-        // TODO: perf: find better way than https://github.com/rolldown/vite/blob/3bf86e3f715c952a032b476b60c8c869e9c50f3f/packages/vite/src/node/plugins/json.ts#L55-L57
-        let str = serde_json::to_string(&str)?;
-        format!("export default /*#__PURE__*/ JSON.parse({str})")
-      } else {
-        format!("export default /*#__PURE__*/ JSON.parse({})", serde_json::to_string(code)?)
-      };
-      return Ok(Some(HookTransformOutput {
-        code: Some(normalized_code),
-        module_type: Some(ModuleType::Js),
-        ..Default::default()
-      }));
+
+    if self.stringify != JsonPluginStringify::False {
+      if self.named_exports && code.trim_start().starts_with('{') {
+        let parsed = serde_json::from_str::<Value>(code)?;
+        let parsed =
+          parsed.as_object().expect("should be object because the value starts with `{`");
+
+        let mut code = String::new();
+        let mut default_object_code = "{\n".to_owned();
+        for (key, value) in parsed {
+          if rolldown_utils::ecmascript::is_validate_assignee_identifier_name(key) {
+            code += &format!("export const {key} = {};\n", &serialize_value(value)?);
+            default_object_code += &format!("  {key},\n");
+          } else {
+            // TODO: escape key
+            default_object_code += &format!("  \"{key}\": {},\n", &serialize_value(value)?);
+          }
+        }
+        default_object_code += "}";
+
+        code += &format!("export default {default_object_code};\n");
+        return Ok(Some(HookTransformOutput {
+          code: Some(code),
+          map: Some(SourceMap::default()),
```

Comment:
The JS plugin has `map: { mappings: '' }`. Is it correct to use `SourceMap::default()` for that?

Reply:
@underfin cc

Reply:
Yeah. It could be using it.